pygmt/clib/session.py

```diff
@@ -1946,6 +1951,70 @@ def virtualfile_to_dataset(
             return result.to_numpy()
         return result  # pandas.DataFrame output
 
+    def virtualfile_to_raster(
+        self,
+        vfname: str,
+        kind: Literal["grid", "image", "cube", None] = None,
+        outgrid: str | None = None,
+    ) -> xr.DataArray | None:
+        """
+        Output a raster data stored in a virtual file to an :class:`xarray.DataArray`
+        object.
+
+        The raster data can be a grid, an image or a cube.
+
+        Parameters
+        ----------
+        vfname
+            The virtual file name that stores the result grid.
```

Comment:
Not implemented for image or cube yet, but document it first so we don't forget later.

```suggestion
            The virtual file name that stores the result grid/image/cube.
```